### PR TITLE
Manage a single identity provider for an organization

### LINKED ISSUE #28272: Manage a single identity provider for an organization

### PR DESCRIPTION
**Pull request by vramik**
Closes #28272

### CODE REVIEW

File: testsuite/integration-arquillian/tests/pom.xml [hunk 1]
```diff
@@ -1600,6 +1600,10 @@
                             <groupId>com.google.guava</groupId>
                             <artifactId>guava</artifactId>
                         </exclusion>
+                        <exclusion>
+                            <groupId>org.codehaus.groovy</groupId>
```
Comment (pedroigor) [inline testsuite/integration-arquillian/tests/pom.xml:1604]: Is this intended?

**Review by pedroigor — DISMISSED** [review]

File: testsuite/integration-arquillian/tests/pom.xml (hunk 1 above)
Comment (vramik) [reply testsuite/integration-arquillian/tests/pom.xml:1604]: It should be probably sent as a separate PR. There is version collision with `groovy-all` artifact in base testsuite when running with `auth-server-quarkus-embedded`.

Version `4.0.16` via `keycloak-juni5` and another is `2.4.10` brought by `creaper-commands`.

I'll create another issue and send a separate PR with it.

File: services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProviderResource.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            String errorMessage;
+            try {
+                if (organizationProvider.addIdentityProvider(organization, identityProvider)) {
+                    return response;
+                }
+                errorMessage = "Assigning the Identity provider with the organization was not succesful.";
+            } catch (ModelException me) {
+                errorMessage = me.getMessage();
+            }
+            response.close();
```
[the remark this answers is not in the document]
Comment (vramik) [reply services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProviderResource.java:99]: At this point we know we will throw an exception and return `BAD_REQUEST` on the next line. I assumed we should close the original response returned from `IdentityProvidersResource`, isn't that the case?

File: testsuite/integration-arquillian/tests/pom.xml (hunk 1 above)
Comment (vramik) [reply testsuite/integration-arquillian/tests/pom.xml:1604]: https://github.com/keycloak/keycloak/issues/28555

**Review by pedroigor — APPROVED** [review]

**Pull request merged**